### PR TITLE
Refactor all compilers to work on a DocumentationSet

### PR DESCRIPTION
**Pull request by mvriel**
Since the compilers work on files and indexes on a DocumentationSet, the entirety of them should really interact with a DocumentationSet and not a project

### CODE REVIEW

File: src/phpDocumentor/Compiler/Pass/Debug.php [hunk 1]
```diff
@@ -48,11 +48,11 @@ public function getDescription(): string
         return 'Analyze results and write report to log';
     }
 
-    public function __invoke(ProjectInterface $project): ProjectInterface
+    public function __invoke(DocumentationSetDescriptor $documentationSet): DocumentationSetDescriptor
     {
-        $this->analyzer->analyze($project);
+        $this->analyzer->analyze($documentationSet);
```
Comment (jaapio) [inline src/phpDocumentor/Compiler/Pass/Debug.php:53]: This should only run on apiSet?

**Review by jaapio — COMMENTED** [review]
I'm missing some checks on ApiSetDescriptor in some of the compiler passes. Did I overlook something?

**Comment by mvriel** [comment]
> I'm missing some checks on ApiSetDescriptor in some of the compiler passes. Did I overlook something?

I have explicitly not added these and tried to keep it as generic as possible. At the moment, guides do not use files and indexes; but I would love to have this a generic mechanism across all Documentation Sets.

At this point, because only API Sets populate these indexes, I have limited the checks on API Sets only where it really is about API Sets. You can find a few of these in the compiler passes themselves

File: src/phpDocumentor/Compiler/Pass/Debug.php (hunk 1 above)
Comment (mvriel) [reply src/phpDocumentor/Compiler/Pass/Debug.php:53]: The analyzer itself does varying things depending on the type of documentation set

**Pull request merged**